modules/extension/wmts/src/main/java/org/geotools/ows/wmts/request/AbstractGetTileRequest.java

```diff
@@ -351,12 +351,15 @@ private TileMatrixSet selectMatrixSet() throws ServiceException, RuntimeExceptio
             }
         }
 
-        // If the layer provides the requested CRS, use it
+        // See if the layer supports the requested SRS. Matching against the SRS rather than the
+        // full CRS will avoid missing matches due to Longitude first being set on the request
+        // CRS and therefore minimise transformations that need to be performed on the received tile
         for (TileMatrixSet matrixSet : capabilities.getMatrixSets()) {
 
             CoordinateReferenceSystem matrixCRS = matrixSet.getCoordinateReferenceSystem();
-
-            if (CRS.equalsIgnoreMetadata(requestCRS, matrixCRS)) { // matching SRS
+            String requestSRS = CRS.toSRS(requestCRS);
```

Comment:
No need to convert it at every round in the loop, could be done outside.

Reply:
Agreed, moved to outside the loop